async-context/README.md

```diff
@@ -0,0 +1,375 @@
+            myLoggingAPI(`i is ${i}`);
+            ++i;
+        } else {
+            clearInterval(interval);
+        }
+    }, 1000);
+}
+
+f1();
+```
+
+Most JS programmers should have an intuitive understanding of what happens when the code above is executed.  We'll build off of this example going forward.  Implicit in this example is an undefined concept of "async context".  We'll build stronger definitions later on, but for now, we'll use "async context" to refer to JS programmers' colloquial understanding of asynchonrous function calls.
+
+There are three interesting constructs in the above example:
+  - First, the function `f2` is a function created in one "async context" and passed through an API.  When `f2` is invoked later, it is a "logical continuation" of the "context" in which it was created.
+  - Second, the function `setInterval` takes a function as a parameter, and invokes that function later during program execution.  Note that when this parameter is invoked, the call to `setInterval` has completed - i.e., `setInterval` is not on the stack when `f2` is invoked. 
+  - Third, `f2` is invoked precisely twice.  These two invocations are distinct.
+
+Let's give these three concepts some names and some initial definitions:
+
+   - A **Continuation** is a JavaScript function that retains a link to the "async context" in which it is created.  Upon invocation, it will establish a new "async context", and will provide a "logical continuation" of the "async context" in which it was created.  In the example above, `f2` is a **Continuation**.
+   - A **Continuation Point** - is a function that accepts a **Continuation** as a parameter.  This logically represents an async boundary; functions passed to a **Continuation Point** are invoked at some later point in time.  In our example above, `setInterval` is a **Continuation Point**. 
+  -  A **Continuation Invocation** is a specific invocation of a `continuation`; more precisely, a **Continuation Invocation** is the *period of time* where a `continuation` frame is on the stack.  As soon as the continuation frame pops off the stack, then that **Continuation Invocation** is completed. Note that a `continuation` instance can have more than one **Continuation Invocation** instances associated with it.  In our code sample above, there are precisely two **Continuation Invocation** instances associated with the `continuation` f2. 
+
+## A lower-level view
+
+At runtime we can view the example above as two distinct call stacks at specific points in program execution.  These look something like this:
+
+```
+---------------------------
+|   setInterval()         |
+---------------------------
+|   function f1()         |
+---------------------------
+|    ...host code...      |
+---------------------------
+|    ...host code...      |
+---------------------------
+|    host function A()    |
+---------------------------
+```
+
+and
+
+```
+---------------------------
+| function x()            |
```

Comment:
I think if you are going to add x(), its probably a good idea to include in the example myLoggingAPI() showing it calling x().

Otherwise maybe just remove x() from the stack?

Reply:
fixed for clarity